Fix java docs (#18479)
visit-dav/visit: src/doc/java_client/index.rst

    Is forwarded to VisIt_ launcher to make VisIt_ prompt for passwords in the console/terminal from which Java was run.

All of the examples are available from the top (``.``) directory of the untarred ``jvisit<version>.tar.gz``.

Java Classes Documentation
--------------------------

There is a ``docs`` subdirectory from the untarred ``jvisit<version>.tar.gz`` with a bunch of ``.html`` files.
If you point a web browser at ``index.html`` there, you can find a lot of documentation on the various Java classes available and used in these examples.

Running the Examples
--------------------

Assuming the current working directory is the directory where ``jvisit<version>.tar.gz`` was untarred, to run an example named ``Example.class``, do the following...

.. code-block:: shell

    java -cp .:visit.jar Example -stay -path /path/to/visit/bin -datapath /path/to/silo/data/dir/

.. note:: The Java program to run, here ``Example``, does not include the ``.class`` extension when it is specified on the command line to ``java``.

   The trailing slash for the ``-datapath`` argument is required.

   All arguments *before* the name of the Java program to run are arguments to ``java``.

>    All arguments *after* the name of the Java program to run are arguments to the program.

Reviewer: ```suggestion
   All arguments *after* the name of the Java program to run are arguments to the program, or to VisIt_.
   Any arguments not consumed by the Java program are forwarded to VisIt_.
```

The ``RunViewer`` program does much of its initialization of the ViewerProxy, the main class for controlling VisIt_'s viewer, in its ``run()`` method.
The actual VisIt_-related work, however, is defined in the ``work()`` method and is overridden in subclasses to perform different VisIt_ functionality.
This program's ``work()`` method opens the VisIt_ globe.silo database, sets some annotation properties, and makes a Pseudocolor plot and a Mesh plot.
After the plots are drawn, the program changes the plot variables a couple of times and saves out images for each of the plotted variables. 

.. container:: collapsible

    .. container:: header

        **Show/Hide Code for** ``RunViewer``

    .. literalinclude:: ../../java/RunViewer.java
       :language: Java

Controlling lighting
~~~~~~~~~~~~~~~~~~~~

This example program is based on the ``RunViewer`` example program and it shows how to modify lighting in VisIt_.
The basic procedure is to obtain a handle to the state object that you want to modify, in this case, ``LightList`` and then modify the state object and call its Notify() method to send the changed object back to VisIt_'s viewer.

> Once the changed object has been sent back to VisIt_'s viewer, you call a method from ViewerMethods that tells VisIt to apply the sent object to its internal state. 

Reviewer: ```suggestion
Once the changed object has been sent back to VisIt_'s viewer, you call a method from ViewerMethods that tells VisIt_ to apply the sent object to its internal state. 
```

.. container:: collapsible

    .. container:: header

        **Show/Hide Code for** ``TryLighting``

    .. literalinclude:: ../../java/TryLighting.java
       :language: Java

Performing queries
~~~~~~~~~~~~~~~~~~

> This example program shows how to use some of VisIt's query capabilities to perform picks and lineouts.

Reviewer: ```suggestion
This example program shows how to use some of VisIt_'s query capabilities to perform picks and lineouts.
```

.. container:: collapsible

    .. container:: header

        **Show/Hide Code for** ``GetMetaData``

    .. literalinclude:: ../../java/GetMetaData.java
       :language: Java

Controlling annotations
~~~~~~~~~~~~~~~~~~~~~~~

This example program shows how to control various annotation objects via the Java API. 

.. container:: collapsible

    .. container:: header

        **Show/Hide Code for** ``TryAnnotations``

    .. literalinclude:: ../../java/TryAnnotations.java
       :language: Java

Making host profiles
~~~~~~~~~~~~~~~~~~~~

This program shows how to create a host profile, add it to the host profile list, and send it to the viewer.
The program then goes on to access data on the remote computer, making use of the host profile that was created.
Additional options such as how to launch the engine in parallel could be added to the host profile.
Also, more profiles could be added to the host profile list before sending it to the viewer. 

.. container:: collapsible

    .. container:: header

        **Show/Hide Code for** ``MakeHostProfile``

    .. literalinclude:: ../../java/MakeHostProfile.java
       :language: Java

> Opening the visIt GUI from Java

Reviewer: ```suggestion
Opening the VisIt_ GUI from Java
```

``-nopty``:
    Is forwarded to VisIt_ launcher to make VisIt_ prompt for passwords in the console/terminal from which Java was run.

All of the examples are available from the top (``.``) directory of the untarred ``jvisit<version>.tar.gz``.

Java Classes Documentation
--------------------------

There is a ``docs`` subdirectory from the untarred ``jvisit<version>.tar.gz`` with a bunch of ``.html`` files.
If you point a web browser at ``index.html`` there, you can find a lot of documentation on the various Java classes available and used in various of these examples.

Running the Examples
--------------------

Assuming the current working directory is the directory where ``jvisit<version>.tar.gz`` was untarred, to run an example named ``Example.class``, do the following...

.. code-block:: shell

    java -cp .:visit.jar Example -stay -path /path/to/visit/bin -datapath /path/to/silo/data/dir/

.. note:: The Java program to run, here ``Example``, does not include the ``.class`` extension when it is specified on the command line to ``java``.
   The trailing slash for the ``-datapath`` argument is required.

>    From within a development build of VisIt_, the ``-path`` option and its argument may be replaced with ``-dv``.

Reviewer: "-dv" is not the same as "-path". "-dv" means you are working from a development version of VisIt, and currently the examples then expect that "../bin" respective to the location of the java example is where "visit" executable is located.  That is certainly not the case on Windows, and may not be the case on Linux.   I think that it is better to say that "-dv" without "--path" assumes "../bin" is where the visit executable is located, and that "--path" should be used if that is not the case.

Author: Ok, I did basically that but I did it up where `-dv` was first documented in the field list.

Additional options such as how to launch the engine in parallel could be added to the host profile.
Also, more profiles could be added to the host profile list before sending it to the viewer. 

.. container:: collapsible

    .. container:: header

        **Show/Hide Code for** ``MakeHostProfile``

    .. literalinclude:: ../../java/MakeHostProfile.java
       :language: Java

Opening the visIt GUI from Java
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

This program shows how to start the VisIt_ **GUI** from within your Java application.
By altering the arguments passed to the ``OpenClient()`` method, you could launch other VisIt_ clients too.
A VisIt_ client is a program that uses the ``ViewerProxy`` class to control the viewer.
Examples of VisIt_ clients are: VisIt_'s **GUI**, VisIt_'s Python interface (**CLI**), and any program that uses the VisIt_ Java interface. 

The important part of this code is the call to the ``OpenClient()`` method.
The ``OpenClient`` method takes 3 arguments: ``clientName``, ``clientProgram``, ``clientArgs``.
The ``clientName`` is the internal name that will be used to identify the client inside of VisIt_.
You can pass any name that you want for this.

> The ``clientProgram`` argment is a string that identifies the executable for your program.

Reviewer: ```suggestion
The ``clientProgram`` argument is a string that identifies the executable for your program.
```

The ``clientArgs`` argument lets you pass command line arguments to your program when it is started.
When you call ``OpenClient()``, the VisIt_ viewer will attempt to launch the specified VisIt_ client and then the client will be attached to VisIt_ and can control the VisIt_ viewer.
Any number of VisIt_ clients can be connected to the VisIt_ viewer.

.. container:: collapsible

    .. container:: header

        **Show/Hide Code for** ``OpenGUI``

    .. literalinclude:: ../../java/OpenGUI.java
       :language: Java

Determining which variables can be plotted
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

This program shows how to open a file and determine which plots can be used with the data from the file.

> Each plot in VisIt_ responds to a certain set of variable types (scalar,vector, and so on).

Reviewer: ```suggestion
Each plot in VisIt_ responds to a certain set of variable types (scalar, vector, and so on).
```

When you open a file, you get a list of variables in the metadata object.
You must match up the variable types supported by a plot and the variables from the metadata in order to determine which plots can accept which variables from the database.
This example program demonstrates a method for doing this comparison.

.. note:: The Java implementation does not offer a ``GetVariableTypes`` method in the plugin interface as it should.

>    This is an oversight that may be corrected in a future version of VisIt.

Reviewer: ```suggestion
   This is an oversight that may be corrected in a future version of VisIt_.
```
